packages/patternfly-3/patternfly-react/src/components/DateTimePicker/DateComponents/DecadeView.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import times from 'lodash/times';
+import { addYears } from './helpers';
+import { noop } from '../../../common/helpers';
+
+class DecadeView extends React.Component {
+  state = {
+    date: new Date(this.props.date),
+    selectedDate: new Date(this.props.date)
+  };
+  getYearArray = () => {
+    const { date } = this.state;
+    date.setFullYear(Math.floor(date.getFullYear() / 10) * 10);
+    return times(12, i => addYears(date, i).getFullYear());
+  };
+  getPrevDecade = () => {
+    const { date } = this.state;
+    this.setState({ date: addYears(date, -10) });
+  };
+  getNextDecade = () => {
+    const { date } = this.state;
+    this.setState({ date: addYears(date, 10) });
+  };
+  setSelectedYear = year => {
+    const { date } = this.state;
+    date.setFullYear(year);
+    this.props.setSelected(date);
+    this.props.toggleDateView('Y');
+  };
+
+  render() {
+    const { date, selectedDate } = this.state;
+    const currDecade = Math.floor(date.getFullYear() / 10) * 10;
+    const selectedYear = selectedDate.getFullYear();
+    const yearArray = this.getYearArray();
+    return (
+      <div className="datepicker-years">
+        <table className="table-condensed">
+          <thead>
+            <tr>
+              <th className="prev" onClick={this.getPrevDecade}>
```

Comment:
optional, maybe divide it into smaller components?

Reply:
Divided into DecadeViewHeader and DecadeViewTable